File: AvernusGames_Back-End/src/Models/Game.cs
using Microsoft.EntityFrameworkCore;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Avernus_Games_Store.src.Models
{
    //Construtores gerados pelo chat GPT
    public class Game
    {   

        public int Id { get; set; }
        public string? Nome { get; set; }
        public string? Descricao { get; set; }
        public float ValorCompra { get; set; }
        public float Markup { get; set; }  
        public Fornecedor? Fornecedor { get; set; }
        public int FornecedorId { get; set; }
        public Genero? Genero { get; set; }
        public int GeneroId { get; set; }
        public Desenvolvedor? Desenvolvedor { get; set; }
        public int DesenvolvedorId { get; set; }
        public Plataforma? Plataforma { get; set; }
        public int PlataformaId { get; set; }
        public Game(){} // Construtor vazio
        
    
    }
}

File: Testes e coisas prontas/src/app/models/Game.ts
import { Desenvolvedor } from "./Desenvolvedor";
import { Genero } from "./Genero";
import { Plataforma } from "./Plataforma";
import { Produto } from "./Produto";

export class Game extends Produto {
    genero: Genero | null = null;
    generoId: number = 0;
    releaseDate: Date | null = null;
    desenvolvedor: Desenvolvedor | null = null;
    desenvolvedorId: number = 0;
    plataforma: Plataforma | null = null;
    plataformaId: number = 0;
  
    CalcValorVenda(ValorCompra: number, Markup: number): number {
      const ValorVenda: number = ValorCompra * (1 + Markup);
      const taxaImposto: number = 0.15; // 15% de imposto
      const ValorTotal: number = ValorVenda * (1 + taxaImposto);
      return ValorTotal;
    }
  }
  